### ISSUE TITLE
what is the direction_loss used to?

### ISSUE THREAD

**Issue by JohnHerry** [lucidrains/e2-tts-pytorch#41]
find no description about the direction_loss in the paper, can this loss improve this model? why keep this in the model init params?

**Comment by JohnHerry** [comment]
and what is the "lens" mean in the DurationPredictor forward function param list?  is it the total mel frames of a sample?
and by the way  about the InterpolatedCharacterEmbed,  what is the structure of the so called "Interpolate"? is it means that with MFA we get frames of each phoneme and then repeat the phoneme to the frame count so that the total len of phoneme is the same of the total len of mel frames?

**Comment by JohnHerry** [comment]
Did not see any backward of duration_predictor loss and its save checkpoint in the trainer.py;  is it training in the same stage of e2-tts training? or training seperately?

**Comment by JohnHerry** [comment]
training pipeline do not support IterableDataset

**Comment by JohnHerry** [comment]
according to the defination of `HFDataset`  and `collate_fn`,  it seems that each sample can be a mel_spec of Tensor type and a text of strign type.  my dataset process the same way, but the training failed, because of the Accelerate  data_loader.py : 756    
```
batch = concatenate(batches, dim=0)
```  
it seems do not support other datatypes except for Tensor,  that means the input text can not be string, or int as token type,  how did the example training run?

**Comment by JohnHerry** [comment]
What is velocity_consistency_loss?  I see no such keyword in e2-tts paper. and it seems that there need a pretrained consistency_model? where can I get it?

**Comment by clumsyroot** [comment]
> Did not see any backward of duration_predictor loss and its save checkpoint in the trainer.py; is it training in the same stage of e2-tts training? or training seperately?

Why this issue has been closed? Did you figure it out?😵

**Comment by JohnHerry** [comment]
是的，默认代码是用来训练e2主模型的，你可以参考他的trainer.py代码，编写你的dp_trainer.py 来单独训练duration predictor模型。两个模型一起用。 velocity_model是使用了指数移动平均 EMA来做模型参数平滑的，但是实际未必用的上，默认是不需要关它的。      
我对pytorch_lightning不太熟悉，所以开始没太懂。如果您对pytorch_lightning比较熟悉的话，这套代码还是比较容易跟的。
